meta2v2/meta2_backend.c

```diff
@@ -1075,8 +1076,10 @@ meta2_backend_delete_alias(struct meta2_backend_s *m2b,
 					deleted_bean=deleted_bean->next) {
 				cb(u0, deleted_bean->data);
 			}
+			g_slist_free(deleted_beans);
+		} else {
+			_bean_cleanl2(deleted_beans);
```

Comment:
If I understood it correctly. `_bean_cleanl2` is only use during test where the function `cb` is not set and we don't free the data inside the list, isn't it ?

Reply:
The callback takes possession of the list elements, and frees them later. When there is no callback, the elements are not freed, thus we must call `_bean_cleanl2` (or equivalent) to free both the list and the elements.